### ISSUE TITLE
document calico 3.5 upgrade process

### ISSUE THREAD

**Issue by jackfrancis** [Azure/aks-engine#908]
The tl;dr is that some manual cleanup will be required for users upgrading clusters that already have an aks-engine addon-manager-enforce calico spec.

Here's a gist that @ahrkrak provided as a reference for helping to develop a definitive howto doc:

https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgist.githubusercontent.com%2Fsong-jiang%2F5a8cdbd5b79ad379506d7e1b83c4685f%2Fraw%2F4f5f643fe844a92237e8f18fa3068cf0248bb31c%2Faks-engine-calico-3.3.1-delete-after-upgrade.yaml&data=02%7C01%7CJack.Francis%40microsoft.com%7Cad0c17af6dcd402e89b408d6aeed59f7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636888731779430243&sdata=vcuBtL9xz3eh4WW1AXwBhhkxMNOsD6oZw%2BQ9DPRn6M0%3D&reserved=0
